### PR TITLE
cli: print client error cause & add --debug flag

### PR DESCRIPTION
**Pull request by danielnorberg**
Make the cli print the message of any encountered `ClientErrorException` and its root cause.

On unknown host, e.g. due to `STYX_CLI_HOST` typo:
```
Client error: Request failed: GET http://syx.spotify.net/api/v3/status/activeStates: UnknownHostException: syx.spotify.net: nodename nor servname provided, or not known
```

On connection failure due to e.g. network problem:
```
Client error: Request failed: GET http://styx.foo.net:4711/api/v3/status/activeStates: ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:4711
```

On http protocol error, e.g. talking to something unexpected:
```
Client error: Request failed: GET http://styx.foo.net:4711/api/v3/status/activeStates: ProtocolException: Unexpected status line: deadbeef
```

The `--debug` flag causes the full stack traces to be printed for any
error caught by the cli. E.g.:

```
com.spotify.styx.client.ClientErrorException: Request failed: GET http://styx.foo.net:4711/api/v3/status/activeStates: ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:4711
	at com.spotify.styx.client.StyxApolloClient.lambda$executeRequest$12(StyxApolloClient.java:452)
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:822)
	at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:797)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at com.spotify.apollo.http.client.TransformingCallback.onFailure(TransformingCallback.java:58)
	at com.squareup.okhttp.Call$AsyncCall.execute(Call.java:185)
	at com.squareup.okhttp.internal.NamedRunnable.run(NamedRunnable.java:33)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Request Request{method=GET, url=http://localhost:4711/api/v3/status/activeStates, tag=null} failed
	at com.spotify.apollo.http.client.TransformingCallback.onFailure(TransformingCallback.java:57)
	... 5 more
Caused by: java.net.ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:4711
	at com.squareup.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:143)
	at com.squareup.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
	at com.squareup.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
	at com.squareup.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
	at com.squareup.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
	at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
	at com.squareup.okhttp.Call.getResponse(Call.java:286)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:243)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:205)
	at com.squareup.okhttp.Call.access$100(Call.java:35)
	at com.squareup.okhttp.Call$AsyncCall.execute(Call.java:171)
	... 4 more
```

An alternative to #502

### CODE REVIEW

File: styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java [hunk 1]
```diff
@@ -305,7 +314,9 @@ private void run() {
           throw CliExitException.of(ExitStatus.ApiError);
         }
       } else if (cause instanceof ClientErrorException) {
-        cliOutput.printError("Client error: " + cause.getMessage());
+        final Throwable rootCause = Throwables.getRootCause(cause.getCause());
```
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java]: No need to cause.getCause() right?

**Comment by ulzha** [comment]
:+1:

File: styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java (hunk 1 above)
Comment (honnix) [reply styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java]: I meant `final Throwable rootCause = Throwables.getRootCause(cause.getCause());` -> `final Throwable rootCause = Throwables.getRootCause(cause);`

Comment (danielnorberg) [reply styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java]: I tried concatenating all the causes but that produced too much redundant and verbose messages for my taste, so I went straight for the important part, the root cause.

**Pull request merged**